src/samples/BotToBot/Bot1/AdapterBotResponseHandler.cs

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Agents.BotBuilder;
+using Microsoft.Agents.Core.Models;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Collections.Generic;
+using Microsoft.Agents.Connector.Types;
+using System;
+using Microsoft.Agents.Client;
+using Microsoft.Agents.Authentication;
+
+namespace Bot1
+{
+    /// <summary>
+    /// Routes Bot response to the Adapter incoming pipeline.  This is the same route a bot normally gets incoming Activities.
+    /// </summary>
+    /// <remarks>
+    /// The default method a Bot responds is Activity.DeliverMode == `normal`.  This is an asynchronous response via an 
+    /// HTTP POST to the endpoints defined by ChannelApiController.
+    /// 
+    /// <see cref="IChannelApiHandler"/> is all of the Connector API endpoints.  This implementation is just handling
+    /// the Send/Reply from the other bot.
+    /// 
+    /// This implementation will send a custom Event to the Adapter, and the AgentApplication can add a route for
+    /// <see cref="AdapterBotResponseHandler.BotResponseEventName"/>.  The Event Activity.Value will be an instance of <see cref="BotResponse"/>.
+    /// </remarks>
+    /// <remarks>
+    /// This implementation does not handle any of the other Connector API endpoints.
+    /// </remarks>
+    public class AdapterBotResponseHandler : IChannelApiHandler
```

Comment:
Could we create a concrete abstract version of this? 
Then override only what we need so we do not have fully implement the interface here in customer code?

Reply:
I imagine this would move to core and be the default handler.  Also, I'm thinking there would be an AgentApplicat.BotResponse object they could add handlers for.  i.e., I would want all this out of the bot and hidden.

Reply:
If in AgentApplication though, I think this means we'd likely always want AddChannelHost done.  Otherwise it would be inert.

Reply:
I wouldn't expect they'd need to change the behavior in this in regard to how the bot response is routed to AgentAppliation.

The remaining endpoints would just route to an On* handler the bot can optionally add (but probably won't).

```
class BotResponseFeature
{
     OnBotResponse(RouteHandler handler);

     OnDeleteActivity(handler);
     OnGetActivityMembers(handler);
    ... and so on
}
```

Then:

```
class AgentApplication
{
     BotResponseFeature BotResponses {get;}
}
```

Then:

```
class MyBot : AgentApplication
{
      MyBot ()
     {
          BotResponses.OnBotResponse(MyBotResponseHandler);
     }

     private Task  MyBotResponseHandler(...)
     {
           // do stuff with the Bot2 response
     }
}
```